spark/dl/src/main/scala/org/apache/spark/ml/DLEstimator.scala

```diff
@@ -84,14 +119,17 @@ class DLEstimator[@specialized(Float, Double) T: ClassTag](
   protected override def internalFit(
       featureAndLabel: RDD[(Seq[AnyVal], Seq[AnyVal])]): DLModel[T] = {
     val batches = toMiniBatch(featureAndLabel)
-    val dataset = DataSet.rdd(batches)
 
-    val optimizer = Optimizer(model, dataset, criterion)
-      .setOptimMethod(new LBFGS[T]())
+    if(!isDefined(optimMethod)) {
+      set(optimMethod, new LBFGS[T])
```

Comment:
in Optimizer, SGD is used by default, do we need to keep consistent or any special case here we should use LBFGS ?